bagit-pax/pitt_pax_v2.py

```diff
@@ -17,6 +17,34 @@
 import botocore
 from concurrent.futures import ThreadPoolExecutor, as_completed
 
+#boolean to decide whether script will be automatic or not
+isAutomated = False
+global first_step
+global c_f_input
+global send_to_s3
+
+#check three parameters to make sure values are correct
+def checkParameters(first_step, c_f_input, send_to_s3):
+    if first_step != "1" and first_step != "2":
+        print("first paramter needs to be 1 or 2")
+        return False
+    if c_f_input != "ALL" and c_f_input != "QUIT" and not c_f_input.isdigit():
+        print("second paramter needs to specify which containers (ALL,1,2,3,QUIT)")
+        return False
+    if send_to_s3 != "1" and send_to_s3 != "0":
+        print("third param needs to be 1 or 0")
+        return False
+    return True
+
+if len(sys.argv) != 4: print("running script without parameters means input will be required later on")
+else:
+    first_step = sys.argv[1]
+    c_f_input = sys.argv[2]
+    send_to_s3 = sys.argv[3] 
+    isAutomated = checkParameters(first_step, c_f_input, send_to_s3)

```

Comment:
Some interesting side effects introduced here.

The user input for `first_step`, et.al., is retained even if `checkParameters()` determines it to be invalid.  You consider this for each subsequent usage by using a `if` conditional built on `isAutomated`, but will each future developer be similarly aware of this side effect?

Reply:
maybe doing something within the `checkParameters` method that doesn't retain the variables if it's false?